src/nsntrace.c

```diff
@@ -245,7 +253,9 @@ netns_main(void *arg) {
 	} else if (child_pid > 0) { /* parent - tracer */
 		struct timespec timeout = { 0 };
 
-		waitpid(child_pid, NULL, 0);
+		waitpid(child_pid, &status, 0);
+		ret = WIFEXITED(status) ? WEXITSTATUS(status) : EXIT_FAILURE;
```

Comment:
Maybe these can be converted to if-statements if you respin this with new commit messages?